### PR TITLE
adds a --pixelaspect %f to oiiotool that intelligently scales the image in X or Y to match the given pixel aspect ratio

### PR DESCRIPTION
**Pull request by mjmvisser**
As described in issue #113. I changed the name to pixelaspect to reflect more accurately what the action does.

e.g., given a 16x32 image with a PAR of 2.0, --pixelaspect 1.0 will result in a 32x32 image with a PAR of 1.0.

This is useful when converting non-square-pixel images to formats that do not support non-square pixels.

Tested with non-square pixel images from here:
http://www.schaik.com/pngsuite/pngsuite_cnk_png.html

### CODE REVIEW

File: src/oiiotool/oiiotool.cpp [hunk 1]
```diff
@@ -2572,6 +2572,97 @@ action_fit (int argc, const char *argv[])
 
 
 
+static int
+action_pixelaspect (int argc, const char *argv[])
+{
+    if (ot.postpone_callback (1, action_fit, argc, argv))
```
Comment (lgritz) [inline src/oiiotool/oiiotool.cpp]: action_pixelaspect!  (not fit)

File: src/oiiotool/oiiotool.cpp [hunk 2]
```diff
@@ -2572,6 +2572,97 @@ action_fit (int argc, const char *argv[])
...
+        return 0;
+    Timer timer (ot.enable_function_timing);
+    bool old_enable_function_timing = ot.enable_function_timing;
+    ot.enable_function_timing = false;
+    string_view command = ot.express (argv[0]);
+
+    float new_paspect = Strutil::from_string<float> (argv[1]);
```
Comment (lgritz) [inline src/oiiotool/oiiotool.cpp]: ot.exrpress (argv[1])

That's what expands any command line expressions.

**Comment by lgritz** [comment]
Very close to LGTM, nothing I can't fix on my end as I merge. I'll make a few minor fixes to it and commit shortly.

**Comment by lgritz** [comment]
I'm toying with the idea of renaming it --scalepixelaspect or something like that. I'm slightly worried that --pixelaspect will be confusing: does it rescale the image, or just change the metadata? The name should make it clear that it's not merely a synonym for --attrib PixelAspectRatio ...

You have any thoughts on this, Mark?

**Comment by lgritz** [comment]
I'm ok with the name, though would consider renaming if you think of something better.

I'll merge tomorrow, plus my extremely minor mods, docs, and test case.

**Comment by lgritz** [comment]
Mark, before I merge this, I want to check one more thing with you. This patch appears to always enlarging one dimension to conform to the requested aspect ratio. For example a 256x256 par=1.0 file, given --pixelaspect 2.0, will result in a 256x512 image, not a 128x256. This approach is nicely conservative -- you don't lose any detail that was in the original image. But some people may be surprised to find the file quite a bit bigger than before. I think that your approach is probably the superior one (never lose detail), but I wanted to be extra sure that's the intended behavior before I merge it.

**Comment by mjmvisser** [comment]
Yeah, I went through similar variations on the name before going with --pixelaspect. I decided against using --scalepixelaspect because it doesn't make sense to scale a pixel aspect ratio. For a while it was --setpixelaspect, but like you mention, that implies (at least to me, anyway) setting the metadata without modifying the image. I suppose it could be --scaletopixelaspect, but that's pretty long for a command line arg. My compromise was to try and make the --help for it as concise as I could.

You're right, I went with the conservative approach intentionally to avoid losing detail. Maybe we could make that explicit in the --help?

     "--pixelaspect %@ %g", action_pixelaspect, NULL, "Scale up the image's width or height to match the given pixel aspect ratio (options: filter=%s)",

cheers

**Comment by lgritz** [comment]
I'm happy with both the name and the behavior. I'll explain in the documentation about the conservative approach.

**Comment by mjmvisser** [comment]
Awesome, thanks!

**Comment by lgritz** [comment]
Merged into master, along with my very small mods, docs, and tests.

**Pull request closed without merging**